examples/spi-hal-one.rs

```diff
@@ -0,0 +1,97 @@
+// This example is to test the SPI without any external devices,
+// it writes to the MOSI line and logs whatever is received on the MISO line
+//
+// MISO and MOSI should be connected to make the assertions work
```

Comment:
From what I can tell, PA6 and PA7 are ARD_D12 and ARD_D11 on the Nucleo G474 which are next to each other. What do about adding this as a test in https://github.com/stm32-rs/stm32g4xx-hal/blob/main/tests/nucleo-g474_w_jumpers.rs ?

Reply:
I've done that and reorganized the tests a bit, I won't have access to some hardware/tools for a while so could you run the test for me?

Reply:
Sure